Source/WTF/wtf/text/CStringView.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2018 Yusuke Suzuki <utatane.tea@gmail.com>
+ * Copyright (C) 2024 Apple Inc. All Rights Reserved.
+ * Copyright (C) 2025 Comcast Inc.
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <span>
+#include <string>
+#include <type_traits>
+#include <wtf/Compiler.h>
+#include <wtf/Forward.h>
+#include <wtf/HashFunctions.h>
+#include <wtf/StdLibExtras.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/SuperFastHash.h>
+
+namespace WTF {
+
+class PrintStream;
+class String;
+
+class CStringView final {
+public:
+    constexpr operator const char*() const LIFETIME_BOUND { return m_charactersWithNullTerminator.data(); }
+
+    static CStringView fromNullTerminatedStringUnsafe(const char* string LIFETIME_BOUND)
+    {
+        if (!string)
+            return CStringView();
+        return CStringView { unsafeMakeSpan(string, std::char_traits<char>::length(string) + 1) };
+    }
+
+    static CStringView fromLiteralUnsafe(const char* string LIFETIME_BOUND)
+    {
+        ASSERT_UNDER_CONSTEXPR_CONTEXT(string);
+        return CStringView::fromNullTerminatedStringUnsafe(string);
+    }
+
+    WTF_EXPORT_PRIVATE static CStringView fromString(const String& string LIFETIME_BOUND);
+
+    WTF_EXPORT_PRIVATE void dump(PrintStream& out) const;
+
+    CStringView() = default;
+    constexpr CStringView(std::nullptr_t)
+        : CStringView()
+    { }
+    CStringView(ASCIILiteral literal LIFETIME_BOUND)
```

Comment:
This is insufficient. We must have constructors that take types other than ASCIILiteral, otherwise this class can't fulfill its purpose.

Reply:
I don't follow. Such as? We can create static functions to create from C Literals.